In [5]:
import pandas as pd 
import numpy as np

## Load the Dataset

In [6]:
train_data = pd.read_csv("train.csv")

train_data.sample(5)

PassengerId  Survived  Pclass  \
334          335         1       1   
90            91         0       3   
255          256         1       3   
649          650         1       3   
121          122         0       3   

                                                  Name     Sex   Age  SibSp  \
334  Frauenthal, Mrs. Henry William (Clara Heinshei...  female   NaN      1   
90                                Christmann, Mr. Emil    male  29.0      0   
255            Touma, Mrs. Darwis (Hanne Youssef Razi)  female  29.0      0   
649                    Stanley, Miss. Amy Zillah Elsie  female  23.0      0   
121                         Moore, Mr. Leonard Charles    male   NaN      0   

     Parch     Ticket      Fare Cabin Embarked  
334      0   PC 17611  133.6500   NaN        S  
90       0     343276    8.0500   NaN        S  
255      2       2650   15.2458   NaN        C  
649      0   CA. 2314    7.5500   NaN        S  
121      0  A4. 54510    8.0500   NaN        S

In [7]:
# test data
test_data = pd.read_csv("test.csv")
test_data.sample(5)

PassengerId  Pclass                                               Name  \
173         1065       3                                   Torfa, Mr. Assad   
66           958       3                            Burns, Miss. Mary Delia   
305         1197       1  Crosby, Mrs. Edward Gifford (Catherine Elizabe...   
211         1103       3                                  Finoli, Mr. Luigi   
26           918       1                       Ostby, Miss. Helene Ragnhild   

        Sex   Age  SibSp  Parch              Ticket     Fare Cabin Embarked  
173    male   NaN      0      0                2673   7.2292   NaN        C  
66   female  18.0      0      0              330963   7.8792   NaN        Q  
305  female  64.0      1      1              112901  26.5500   B26        S  
211    male   NaN      0      0  SOTON/O.Q. 3101308   7.0500   NaN        S  
26   female  22.0      0      1              113509  61.9792   B36        C

In [12]:
train_data.describe

<bound method NDFrame.describe of      PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                 

In [13]:
test_data.describe

<bound method NDFrame.describe of      PassengerId  Pclass                                          Name  \
0            892       3                              Kelly, Mr. James   
1            893       3              Wilkes, Mrs. James (Ellen Needs)   
2            894       2                     Myles, Mr. Thomas Francis   
3            895       3                              Wirz, Mr. Albert   
4            896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
..           ...     ...                                           ...   
413         1305       3                            Spector, Mr. Woolf   
414         1306       1                  Oliva y Ocana, Dona. Fermina   
415         1307       3                  Saether, Mr. Simon Sivertsen   
416         1308       3                           Ware, Mr. Frederick   
417         1309       3                      Peter, Master. Michael J   

        Sex   Age  SibSp  Parch              Ticket      Fare Cabin Embarked 

Now, We have all **891 rows and 12 columns** in train.csv & **418 rows and 11 columns** in test.csv

## Our First Submission

Our Goal is to find patterns in **train.csv** that help us predict whether the passengers in **test.csv** survived.

It might initially feel overwhelming to look for patterns, when there's so much data sort through. So, we'll start simple.

## Let's Find a Pattern
Keep in mind that the sample submission file, **gender_submission.csv**, operates under the assumption that every female passenger survived while all male passengers did not.

In this a sensible initial assumption? We will verify if this trend is consistent with the data found in train.csv

In [8]:
women = train_data.loc[train_data.Sex == 'female']["Survived"]

women

1      1
2      1
3      1
8      1
9      1
      ..
880    1
882    0
885    0
887    1
888    0
Name: Survived, Length: 314, dtype: int64